legwork/source.py

```diff
@@ -486,21 +500,33 @@ def get_snr_stationary(self, t_obs=4 * u.yr, which_sources=None,
             if verbose:
                 print("\t\t{} sources are stationary and eccentric".format(
                     len(snr[ind_ecc])))
-            max_harmonics = self.max_harmonic(self.ecc)
+            harmonics_required = self.harmonics_required(self.ecc)
             harmonic_groups = [(1, 10), (10, 100), (100, 1000), (1000, 10000)]
             for lower, upper in harmonic_groups:
-                harm_mask = np.logical_and(max_harmonics >= lower,
-                                           max_harmonics < upper)
+                harm_mask = np.logical_and(harmonics_required >= lower,
+                                           harmonics_required < upper)
                 match = np.logical_and(harm_mask, ind_ecc)
                 if match.any():
-                    snr[match] = sn.snr_ecc_stationary(m_c=self.m_c[match],
-                                                       f_orb=self.f_orb[match],
-                                                       ecc=self.ecc[match],
-                                                       dist=self.dist[match],
-                                                       t_obs=t_obs,
-                                                       max_harmonic=upper - 1,
-                                                       interpolated_g=self.g,
-                                                       interpolated_sc=self.sc)
+                    hr = upper - 1
+
+                    snr_msh = sn.snr_ecc_stationary(m_c=self.m_c[match],
+                                                    f_orb=self.f_orb[match],
+                                                    ecc=self.ecc[match],
+                                                    dist=self.dist[match],
+                                                    t_obs=t_obs,
+                                                    harmonics_required=hr,
+                                                    interpolated_g=self.g,
+                                                    interpolated_sc=self.sc,
+                                                    ret_max_snr_harmonic=True)
+                    snr[match], msh[match] = snr_msh
```

Comment:
I had to do this to prevent it going over 80 characters otherwise flake would get angry with me